src/Session.php

```diff
@@ -92,6 +92,8 @@ public function open(): void
 
         if ($this->sessionId !== null) {
             session_id($this->sessionId);
+        } else {
+            session_id(session_create_id());
```

Comment:
What about collision check? `session_create_id` don't check collision if called before start session.

Reply:
Any idea?

Reply:
I don't know. In symfony it's not check. May be to make custom session ID creating optional? Use boolean flag or callback or create `SessionIdGeneratorInterface`?

Reply:
I think this would be very non-intuitive for users

Reply:
> I think this would be very non-intuitive for users

Then we must find solutions without collision risk. Otherwise we fix bug for longliving apps, but add new problem with collisions.

Reply:
From PHP documentation:

`session_create_id()` is used to create new session id for the current session. It returns *collision free* session id.

If session is not active, collision check is omitted.

https://www.php.net/manual/en/function.session-create-id.php

Reply:
What about this?

```php
session_start($this->options);
$this->sessionId = session_create_id();
session_write_close();
session_id($this->sessionId);
```

Reply:
> What about this?
> 
> ```
> session_start($this->options);
> $this->sessionId = session_create_id();
> session_write_close();
> session_id($this->sessionId);
> ```

This will entail additional costs on read/write session data.

Reply:
Extra opening a session doesn't sound good to me.

Reply:
There's also https://www.php.net/manual/en/session.configuration.php#ini.session.use-strict-mode. We should take that into account.